### PR TITLE
Fix Trace.times(type='matplotlib') being slow

### PR DESCRIPTION
**Pull request by ahhbjoe**
### What does this PR do?

The current version of obspy.core.trace.Trace.times is very slow for type="matplotlib". This PR speeds it up by ~1000 times.

### Why was it initiated?  Any relevant Issues?

I don't see relevant issues.

### PR Checklist
- [x] Correct base branch selected? `master` for new fetures, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] All tests still pass.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [x] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

File: obspy/core/trace.py [hunk 1]
```diff
@@ -2496,8 +2496,11 @@ def times(self, type="relative", reftime=None):
                 [self.stats.starttime + t_ for t_ in time_array])
         elif type == "matplotlib":
             from matplotlib.dates import date2num
-            time_array = date2num([(self.stats.starttime + t_).datetime
-                                   for t_ in time_array])
+            SECONDSPERDAY = 86400
```
Comment (krischer) [inline obspy/core/trace.py]: Having this as an extra variable is probably not necessary as most people should recognize that numbers and its only used once in the next name.

**Comment by megies** [comment]
Nice first PR there!  :clap:

**Comment by megies** [comment]
Dunno, maybe a short changelog entry might be good to have, but not thaaat important.

**Comment by megies** [comment]
> Please install flake8 and pep8-naming locally and just run $ flake8 filename.py to make sure the formatting corresponds to our rules

Pro-tip: Use a pre-commit hook as a safe guard in the future.. :sunglasses: 
See http://docs.obspy.org/coding_style.html#obspy-coding-style-guide right at the top..

File: obspy/core/trace.py (hunk 1 above)
Comment (ahhbjoe) [reply obspy/core/trace.py]: Thanks for your suggestions. Requested changes have been made.

**Comment by ahhbjoe** [comment]
It seems that CircleCI was checking an older version of my codes. The error messages are "variable 'SECONDSPERDAY' in function should be lowercase" and "trailing whitespace". But both problems have been fixed in the current version.

**Comment by krischer** [comment]
Thanks! AppVeyor fails are unrelated.

**Comment by megies** [comment]
Thanks for your PR, @ahhbjoe. Great job!  :tada:

**Pull request merged**